lib/npo.src.js

```diff
@@ -77,6 +124,15 @@
 			cycle = timer(scheduling_queue.drain);
 		}
 	}
+	
+	function hasCatch(chain) {
+		for (var i = 0; i < chain.length; i++) {
+			if (chain[i].failure) {
+				return true;
+			}
+		}
+		return false;
+	}
```

Comment:
Another thing I'm not sure about in the definition of what's an "unhandled rejection"...

```js
var A = Promise.reject("oops");

var B = A.then(function(){});
var C = A.catch(function(){});
```

Which looks like this:

```
      A
    /   \
  B       C
```

Should the `"oops"` rejection be reported as unhandled, or not? It's handled on the `C` branch, but not on the `B` branch.

In the chrome console, it reports an unhandled rejection. As a matter of fact, if both `B` and `C` don't catch, it reports two unhandled rejections. This seems strange to me, and doesn't fit with my understanding that a single rejection has an "identity" no matter how far it propagates down a chain.